DatadogSessionReplay/Sources/Recorder/Utilities/ImageDataProvider.swift

```diff
@@ -90,14 +100,18 @@ import CryptoKit
 
 private var customHashKey: UInt8 = 11
 fileprivate extension UIImage {
+    private static var associatedObjectQueue = DispatchQueue(label: "com.datadoghq.customHashQueue")
+
     var customHash: String {
-        if let hash = objc_getAssociatedObject(self, &customHashKey) as? String {
+        return UIImage.associatedObjectQueue.sync {
+            if let hash = objc_getAssociatedObject(self, &customHashKey) as? String {
+                return hash
+            }
+
+            let hash = computeHash()
+            objc_setAssociatedObject(self, &customHashKey, hash, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
             return hash
         }
-
-        let hash = computeHash()
-        objc_setAssociatedObject(self, &customHashKey, hash, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
-        return hash
     }
```

Comment:
> And this identifier might be accessed (and lazily computed) from two different places.

TBH, it makes it very hard to read and reason about this code IMO. Depending on who calls this code first, it will cause impact on this or that thread. If tomorrow, we call it on the main thread, the SDK will cause huge impact (queue.sync {} + png compression + md5 computation) on user interface. It can be wrapped by the getter vs side-effects discussion from https://github.com/DataDog/dd-sdk-ios/pull/1585#discussion_r1439549398.

I'd prefer more streamlined design, where `srIdentifier` is set explicitly by respective thread. Other threads should only retrieve it in performant fashion.